File: 0412-fizz-buzz/0412-fizz-buzz.js
/**
 * @param {number} n
 * @return {string[]}
 */
var fizzBuzz = function(n) {
    let array = [];
    let index = 1;
    while (index <= n) {
        let value = index.toString();
        let fizz = "";
        if (index % 3 === 0) fizz += "Fizz";
        if (index % 5 === 0) fizz += "Buzz";
        array.push(fizz !== "" ? fizz : value);
        ++index;
    };
    return array;
};

File: 0412-fizz-buzz/0412-fizz-buzz.py
class Solution(object):
    def fizzBuzz(self, n):
        """
        :type n: int
        :rtype: List[str]
        """
        list = []

        for i in range(1, n + 1):
          fizzCondition = i % 3 == 0
          buzzCondition = i % 5 == 0
          
          if fizzCondition and buzzCondition:
              list.append("FizzBuzz")
          elif fizzCondition:
              list.append("Fizz")
          elif buzzCondition:
              list.append("Buzz")
          else: 
              list.append(str(i))

        return list